### PR TITLE
feat(access-groups): add PATCH /management/v1/access-groups/{id} and migrate the edit modal to RHF + zod + shadcn

### PR DESCRIPTION
**Pull request by ryan-crabbe-berri**
## TLDR

Problem this solves:

- Edit Access Group modal was still antd, unlike the create dialog
- No partial-update route for access groups; PUT accepts and ignores typos
- Two-field edits before the first save silently dropped the second field
- No `/management/v1` mutation exists yet to prove the standard

How it solves it:

- New `PATCH /management/v1/access-groups/{id}`: `{"data"}` envelope, merge patch, 422 on unknown keys
- Errors are RFC 9457 problem documents, declared in OpenAPI per status
- New shadcn edit dialog shares tabbed fields with the create dialog
- Only dirty fields go over PATCH; blank description clears with null
- Fix usePickDirty so RHF actually tracks every dirty field (org settings too)

## User Flow

Before: an admin editing an access group gets an antd modal, and there is no partial-update route

1. They open http://localhost:4000/ui/?page=access-groups, click a group, then Edit Access Group
2. An antd modal opens with General Info, Models, MCP Servers and Agents tabs
3. They save and the whole form goes over PUT https://litellm-domain/v1/access_group/{id}
4. A script sending PATCH https://litellm-domain/management/v1/access-groups/{id} with `{"description": "x"}` gets 404 `{"detail":"Not Found"}`
5. The same script sending PUT https://litellm-domain/v1/access_group/{id} with `{"descripton": "typo"}` gets 200 and nothing changes
6. On the Organization settings form, renaming and clearing models before the first save only sends the rename

After: the same admin gets a shadcn dialog that saves over the new control-plane PATCH

1. They open http://localhost:4000/ui/?page=access-groups, click a group, then Edit Access Group
2. A shadcn dialog opens with the same four tabs, hydrated from the group, Save disabled until something changes
3. They change the description, add an MCP server and an agent across tabs, and save once: a single PATCH https://litellm-domain/management/v1/access-groups/{id} carries exactly those three keys, comes back 200 as `{"data": {...full group...}}`, the page refreshes with the new values, and reopening shows them
4. Blanking the description sends `"description": null`, emptying a list sends `[]`, Cancel or Escape discards edits, a blank name is refused with an inline error on General Info
5. A script sending PATCH https://litellm-domain/management/v1/access-groups/{id} with `{"description": "x"}` gets 200 and only that column changes; `{"descripton": "typo"}` gets 422 `application/problem+json` naming the key; a wrong id gets a 404 problem, a taken name a 409 problem
6. PUT https://litellm-domain/v1/access_group/{id} keeps working unchanged; PATCH on that legacy path stays 405
7. On the Organization settings form, renaming and clearing models before the first save sends both

## Pre-Submission checklist

**Please complete all items before asking a LiteLLM maintainer to review your PR**

- [x] I have added meaningful tests
- [x] My PR passes all CI/CD checks (e.g., lint, format, unit tests)
- [x] My PR's scope is as isolated as possible; it only solves 1 specific problem
- [ ] I have received a Greptile **Confidence Score of at least 4/5** before requesting a maintainer review (Greptile reviews automatically once the PR is opened; only comment `@greptileai` to re-request a review after pushing changes)

## Delays in PR merge?

If you're seeing a delay in your PR being merged, ping the LiteLLM Team on [Slack (#pr-review)](https://join.slack.com/t/litellmossslack/shared_invite/zt-3o7nkuyfr-p_kbNJj8taRfXGgQI1~YyA).

## Screenshots / Proof of Fix

Setup: proxy from the named commit with the freshly built dashboard, master key `sk-1234`, `H='Authorization: Bearer sk-1234'`, `J='Content-Type: application/json'`. Before ran on http://localhost:4188, After on http://localhost:4187. Each side creates a throwaway group with `POST /v1/access_group` (`{"access_group_name":"qa-...","description":"before","access_model_names":["gpt-5.5","gpt-4o-mini"]}`) and `$ID` is the returned `access_group_id`. The dialog screenshots come from the seed group `146b8235-13c5-49d3-a30d-a82f00ef88ff` with one MCP server `qa_github_mcp` and one agent `qa-hello-agent` registered

#### Edit modal

1. Open http://localhost:4188/ui/?page=access-groups, click the group row, click Edit Access Group
2. antd modal, General Info tab

before general

3. Models tab

before models

#### Partial update route

1. `curl -s -i -X PATCH http://localhost:4188/management/v1/access-groups/$ID -H "$H" -H "$J" -d '{"description":"after"}'`
2. `HTTP/1.1 404 Not Found`, `content-type: application/json`, `{"detail":"Not Found"}`
3. `curl -s -w '\nHTTP %{http_code}\n' -X PUT http://localhost:4188/v1/access_group/$ID -H "$H" -H "$J" -d '{"descripton":"typo"}'`
4. The full group with `"description":"before"` then `HTTP 200`: the typo is a silent no-op
5. `curl -s -w '\nHTTP %{http_code}\n' -X PATCH http://localhost:4188/v1/access_group/$ID -H "$H" -H "$J" -d '{"description":"x"}'`
6. `{"detail":"Method Not Allowed"}` then `HTTP 405`
7. `curl -s http://localhost:4188/openapi.json | python3 -c "import sys,json; s=json.load(sys.stdin); print('/management/v1/access-groups/{access_group_id}' in s['paths'], 'ProblemDetail' in s['components']['schemas'], sorted(s['paths']['/v1/access_group/{access_group_id}']))"`
8. `False False ['delete', 'get', 'put']`

#### Multi-field edit before the first save

1. The antd modal always PUTs the whole form, so this case only exists once the dialog sends dirty fields. Where the bug is observable today is the organization settings form, which already uses `pickDirty`: rename plus clear models before the first save sends `{"organization_alias": "acme-2"}` and drops `models`
2. The new `OrgSettingsForm.test.tsx` case "sends every field edited before the first save" fails at this commit with exactly that body

#### Edit modal

The dialog is unchanged since 1e4f66f0fc, where these screenshots were captured; the tip commit only swaps the route it saves to, and the network capture under "Multi-field edit before the first save" below is from the tip

1. Open http://localhost:4187/ui/?page=access-groups, click the group row, click Edit Access Group
2. shadcn dialog, General Info tab, Save disabled while pristine

after general

3. Models tab

after models

4. MCP Servers tab with a server picked

after mcp

5. Agents tab with an agent picked

after agents

6. Blank the name on General Info, switch to Models, click Save Changes: the dialog stays open, jumps back to General Info and shows the error

after validation

#### Partial update route

1. `curl -s -i -X PATCH http://localhost:4187/management/v1/access-groups/$ID -H "$H" -H "$J" -d '{"description":"after"}'`
2. `HTTP/1.1 200 OK`, `content-type: application/json`, `{"data":{"access_group_id":"4667f9c9-bb70-46e6-a215-52b22c26ef59","access_group_name":"qa-mv1-7427","description":"after","access_model_names":["gpt-5.5","gpt-4o-mini"],"access_mcp_server_ids":[],"access_agent_ids":[],"assigned_team_ids":[],"assigned_key_ids":[],"created_at":"2026-08-17T23:10:03.522000Z","created_by":"default_user_id","updated_at":"2026-08-17T23:10:03.591000Z","updated_by":"default_user_id"}}`: only the description moved
3. `curl -s -X PATCH http://localhost:4187/management/v1/access-groups/$ID -H "$H" -H "$J" -d '{"description":null,"access_model_names":null}'`
4. `{"data":{...,"description":null,"access_model_names":[],...}}`: null clears a scalar, a cleared list comes back as `[]`
5. `curl -s -i -X PATCH http://localhost:4187/management/v1/access-groups/$ID -H "$H" -H "$J" -d '{"descripton":"typo"}'`
6. `HTTP/1.1 422 Unprocessable Content`, `content-type: application/problem+json`, `{"type":"urn:litellm:error:invalid-request-body","title":"Invalid request body","status":422,"detail":"descripton: Extra inputs are not permitted"}`
7. Same with `-d '{"access_model_names":"gpt-5.5"}'`: `{"type":"urn:litellm:error:invalid-request-body","title":"Invalid request body","status":422,"detail":"access_model_names: Input should be a valid list"}`
8. `curl -s -i -X PATCH http://localhost:4187/management/v1/access-groups/does-not-exist -H "$H" -H "$J" -d '{"description":"x"}'`
9. `HTTP/1.1 404 Not Found`, `content-type: application/problem+json`, `{"type":"urn:litellm:error:not-found","title":"Not found","status":404,"detail":"Access group 'does-not-exist' not found."}`
10. Create a second group `qa-b-29659`, then `curl -s -i -X PATCH http://localhost:4187/management/v1/access-groups/$ID -H "$H" -H "$J" -d '{"access_group_name":"qa-b-29659"}'`
11. `HTTP/1.1 409 Conflict`, `content-type: application/problem+json`, `{"type":"urn:litellm:error:conflict","title":"Conflict","status":409,"detail":"Access group 'qa-b-29659' already exists."}`
12. `curl -s -o /dev/null -w '%{http_code}\n' -X PUT http://localhost:4187/v1/access_group/$ID -H "$H" -H "$J" -d '{"description":"via put"}'` prints `200`; the same call as `-X PATCH` prints `405`
13. `curl -s http://localhost:4187/openapi.json | python3 -c "import sys,json; s=json.load(sys.stdin); op=s['paths']['/management/v1/access-groups/{access_group_id}']['patch']; print(sorted(op['responses']), list(op['responses']['422']['content']), op['responses']['200']['content']['application/json']['schema'], 'ProblemDetail' in s['components']['schemas'], sorted(s['paths']['/v1/access_group/{access_group_id}']))"`
14. `['200', '403', '404', '409', '422', '500', '503'] ['application/problem+json'] {'$ref': '#/components/schemas/ItemResponse_AccessGroupResponse_'} True ['delete', 'get', 'put']`
15. A key generated with `"user_role":"internal_user"` sending the same PATCH gets the proxy-wide `401` for `/management/v1/*` before the route runs, same as budgets today

#### Multi-field edit before the first save

1. Fresh page load, open the group, Edit Access Group (Save disabled), pick `qa_github_mcp` on MCP Servers, pick `qa-hello-agent` on Agents, change the description on General Info, click Save Changes
2. Network shows one `PATCH http://localhost:4187/management/v1/access-groups/146b8235-13c5-49d3-a30d-a82f00ef88ff` with body `{"description":"Edited on three tabs before the first save","access_mcp_server_ids":["5ff60808-672c-4b18-8ad7-5d4ee2a92f83"],"access_agent_ids":["23100095-07b1-42f0-8501-4921001d83d8"]}`
3. Response `200` `{"data":{"access_group_id":"146b8235-13c5-49d3-a30d-a82f00ef88ff","access_group_name":"prod-models-renamed","description":"Edited on three tabs before the first save","access_model_names":["gpt-5.2","claude-sonnet-5"],"access_mcp_server_ids":["5ff60808-672c-4b18-8ad7-5d4ee2a92f83"],"access_agent_ids":["23100095-07b1-42f0-8501-4921001d83d8"],"assigned_team_ids":[],"assigned_key_ids":[],...}}`; the toast "Access group updated successfully" shows, the dialog closes and the page shows the new description and Models 2, MCP Servers 1, Agents 1 without a reload
4. Reopen: the description and the picked MCP server are there and Save is disabled. Rename to `prod-models-renamed-again` and blank the description, Save: one PATCH with body `{"access_group_name":"prod-models-renamed-again","description":null}`, the heading reads `prod-models-renamed-again` and the description shows the em dash
5. Type into the description then Cancel, reopen: the field is back to the saved (empty) value; same with Escape

## Type

🆕 New Feature
🐛 Bug Fix
🧹 Refactoring

## Caveats (if any)

- First `/management/v1` mutation; the standard's design doc had budgets proving it first
- The legacy PATCH added earlier in this PR is gone; only PUT stays on `/v1/access_group`
- `403` in the route is defense in depth; auth already rejects non-admins on `/management/v1/*` with 401
- `AccessGroupResponse` in the dashboard aliases the generated OpenAPI type (description, created_by, updated_by optional)
- Local basedpyright is flat vs the merge base for reportPrivateUsage and reportUnnecessaryIsInstance; if the CI gate flags them the drift is in the fetched base counts

### Final Attestation

- [x] The tests check the right things, including the edge cases, and regressions in the respective real-world customer use-cases are not possible after this PR

### CODE REVIEW

File: ui/litellm-dashboard/src/app/(dashboard)/access-groups/_components/access-group-edit/AccessGroupEditDialog.test.tsx [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+    expect(screen.getByLabelText("Description")).toHaveValue("Original description");
+  });
+
+  it("cannot be dismissed while a save is pending, then closes once on success", async () => {
+    const user = userEvent.setup();
+    let resolveSave: (value: AccessGroupResponse | undefined) => void = () => {};
+    const patchAccessGroup = vi.fn().mockImplementation(
+      () =>
+        new Promise((resolve) => {
+          resolveSave = resolve;
```
[the remark this answers is not in the document]
Comment (ryan-crabbe-berri) [reply ui/litellm-dashboard/src/app/(dashboard)/access-groups/_components/access-group-edit/AccessGroupEditDialog.test.tsx]: Done in 08a306f198: the fixture now uses Promise.withResolvers, so the resolver is a const and nothing gets reassigned.

**Comment by ryan-crabbe-berri** [comment]
@greptileai re review

**Comment by ryan-crabbe-berri** [comment]
@greptileai please re-review: the partial update moved to PATCH /management/v1/access-groups/{id} with problem+json errors and a 422 on unknown body keys.
